### PR TITLE
New layout format and User data migration capability! Take 2

### PR DESCRIPTION
**Pull request by mominul**
Here I present version 2 of the keyboard method layout format of OBK version 2! :tada:

This layout format uses [Zstandard](https://facebook.github.io/zstd/) compression algorithm for compressing the images and uses a custom variant of Base91 for decoding the binary bytes into textual format. Base91 is 6% more efficient than Base64 which we currently use. As a result, the layout files we distribute altogether have their sizes shrunken to 58% :tada:

With the new layout format, we now support system keyboard layout other than Qwerty(#36) without any [hackery in the parser](https://github.com/OpenBangla/riti/compare/b1b94470aee8b8ed78268b2a4896f7a71a02ca89...8195ea889562875764d29d7f9e573151fbbc4c8a#diff-1cdcf89142d5ae9ce59824fd657223e0) and with additional number keys. I have also added a heavy test for the parser which will prevent further typo bugs(#98) from happening.

I have also changed the names of two layouts:
* Bornona to **Borno**. (This was the real name, I don't know why Avro team had changed it.).
* Munir Optima (uni) to **Munir Optima**.

This PR also adds User data migration capability to OBK. It also includes a converter for the new format. So the next release of OBK will be no longer a breaking version. :tada:

### CODE REVIEW

**Comment by mominul** [comment]
@bdeshi Can you please review the ReadMe and script changes?

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,7 +93,7 @@ set(CPACK_DEBIAN_PACKAGE_DEPENDS "ibus (>= 1.5.1)")
 set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
 set(CPACK_RPM_PACKAGE_LICENSE "GPLv3")
 set(CPACK_RPM_PACKAGE_URL "https://openbangla.github.io/")
-set(CPACK_RPM_PACKAGE_REQUIRES "qt5-qtbase >= 5.9.0, ibus >= 1.5.1, ibus-libs >= 1.5.1")
+set(CPACK_RPM_PACKAGE_REQUIRES "qt5-qtbase >= 5.9.0, ibus >= 1.5.1, ibus-libs >= 1.5.1, zstd >= 1.3.3")
```
Comment (bdeshi) [inline CMakeLists.txt:96]: the zstd binaries are not required. this should be `libzstd`, right?
also, shouldn't `CPACK_DEBIAN_PACKAGE_DEPENDS` list qt5 and zstd dependencies?

Comment (mominul) [reply CMakeLists.txt:96]: > the zstd binaries are not required. this should be `libzstd`, right?
Yeah!
> also, shouldn't `CPACK_DEBIAN_PACKAGE_DEPENDS` list qt5 and zstd dependencies?
Actually the dependencies are automatically included by `dpkg-shlibdeps` as we set `CPACK_DEBIAN_PACKAGE_SHLIBDEPS`.

Comment (bdeshi) [reply CMakeLists.txt]: ah. okay then. other changes are alright.

**Pull request merged**